Docs: remotes guide (#302)
CERT-Polska/mwdb-core: docs/remotes-guide.rst

Remote instances guide
===============

Remote instance introduction
----------------------------------

Remote instances allow us to use external mwdb servers within our local instance.
This gives us the ability to view samples located in the remote instance, as well as pull data from the remote instance to the local instance and push to the remote.

Depending on the permissions we have on the remote instance, the following things are possible for us:

* viewing samples, configs and blobs
* pull action to retrieve an object from a remote instance
* performing a push action to push an object to the remote instance

Remote instance UI are currently limited to the main mwdb views.
Actions affecting the state of the object on a remote instance, such as commenting, adding tags or attributes, were also blocked.

Setting up remote instance
----------------------------------

In order to use the remote instance views, you need the api key associated with the account authorized within it.
If you don't have the api key generated or don't know how to do it, check :ref:`How to use API keys?` for more information.

> With the api key generated already, navigate to the file ``mwdb.ini`` inside ``docker`` folder in main project directory.

Author: ```suggestion
With the api key generated, navigate to the file ``mwdb.ini`` inside ``docker`` folder in main project directory.
```

In this file you should add the definition of the remote instance along with the api key and its url in the form:

.. code-block:: docker
    ...
    remotes = mwdb.cert.pl

    [remote:mwdb.cert.pl]
    url = http://mwdb.cert.pl
    api_key=ey...

> The remotes variable inside this file passes the list of variables to the mwdb configuration files, from which we can browse remote instances within by the local application instance.

Author: ```suggestion
The ``remotes`` variable inside this file passes the list of remotes to the mwdb configuration files, from which we can browse remote instances within by the local application instance.
```

> If we want to define more remote instances, just add more remots in the form

Author: ```suggestion
If we want to define more remote instances, just add more remotes in form
```

At the moment, there are views depending on the object category and the search view.

.. warning::

    The quick query function has been limited because it would refer to the user who owns the api key in the defined remote instance.

By going to the view of a specific object on a remote instance, we will learn detailed information about this object and we will be able to perform two possible actions on it:

* ``Download`` to download the object from a remote instance
* ``Pull`` to load the object into the local mwdb database

The remaining actions that can be performed on the remote object have been limited.

.. image:: ../_static/remote-object-pull.png
   :target: ../_static/remote-object-pull.png
   :alt: remote object view pull

Another way to load an object from a remote instance to a local one is to go to a new view in the local mwdb `` Pull``.
Then select the instance from which you want to pull the object and enter its identifier.

.. image:: ../_static/remote-pull.png
   :target: ../_static/remote-pull.png
   :alt: local remote object view pull

> Similar to the pull action on an object on a remote instance, we can also push an object from a local instance to a remote instance by clicking the appropriate ``push`` button in the local object view.

Author: ```suggestion
Similar to the pull object action in remote instance, we can also push an object from local instance to remote by clicking the appropriate ``push`` button in the local object view.
```

.. image:: ../_static/remote-push.png
   :target: ../_static/remote-push.png
   :alt: remote object view push

> After clicking on the button, a window will appear in which you will have to select the remote instance to which you want to push the object.

Author: ```suggestion
After clicking on that button, a window will appear in which you will have to select the remote instance to which you want push the object.
```
